aws_advanced_python_wrapper/host_monitoring_v2_plugin.py

```diff
@@ -0,0 +1,506 @@
+        except InterruptedError:
+            pass
+        except Exception:
+            logger.debug("HostMonitorV2.ExceptionDuringMonitoringStop", self._host_info.host)
+
+        logger.debug("HostMonitorV2.StopMonitoringThreadNewContext", self._host_info.host)
+
+    def run(self) -> None:
+        logger.debug("HostMonitorV2.StartMonitoringThread", self._host_info.host)
+
+        try:
+            while not self.is_stopped:
+                if self._active_contexts.empty() and not self._is_unhealthy:
+                    sleep(self._THREAD_SLEEP_NANO // 1_000_000_000)
+                    continue
+
+                status_check_start_time_ns: float = self.get_current_time_ns()
+                is_valid: bool = self.check_connection_status()
+                status_check_end_time_ns: float = self.get_current_time_ns()
+
+                self._update_node_health_status(is_valid, status_check_start_time_ns, status_check_end_time_ns)
+
+                temp_active_contexts = []
+                while not self._active_contexts.empty():
+                    monitor_context_weak_ref = self._active_contexts.get()
+                    if self.is_stopped:
+                        break
+
+                    if monitor_context_weak_ref is None:
+                        continue
+
+                    monitor_context = monitor_context_weak_ref()
+
+                    if monitor_context is None:
+                        continue
+
+                    if self._is_unhealthy:
+                        # Kill Connection
+                        monitor_context.set_node_unhealthy()
+                        connection_to_abort = monitor_context.get_connection()
+                        monitor_context.set_inactive()
+                        if connection_to_abort is not None:
+                            self.abort_connection(connection_to_abort)
+                            if self._aborted_connection_counter is not None:
+                                self._aborted_connection_counter.inc()
+                    elif monitor_context.is_active():
+                        temp_active_contexts.append(monitor_context)
+
+                for active_context in temp_active_contexts:
+                    self._active_contexts.put(weakref.ref(active_context))
+
+                delay_ns = self._failure_detection_interval_ns - (status_check_end_time_ns - status_check_start_time_ns)
+                if delay_ns < self._THREAD_SLEEP_NANO:
+                    delay_ns = self._THREAD_SLEEP_NANO
+
+                sleep(delay_ns // 1_000_000_000)
+
+        except InterruptedError:
+            pass
+        except Exception:
+            logger.debug("HostMonitorV2.ExceptionDuringMonitoringStop", self._host_info.host)
+        finally:
+            self.stop()
+            if self._monitoring_connection is not None:
+                try:
+                    self._monitoring_connection.close()
+                except AwsWrapperError:
+                    pass
+
```

Comment:
In JDBC we log a message when the run thread exits, can we do the same?
https://github.com/aws/aws-advanced-jdbc-wrapper/blob/e8a99dfd5773ded759384cf66064144d7294b472/wrapper/src/main/java/software/amazon/jdbc/plugin/efm2/MonitorImpl.java#L332